### PR TITLE
[BinaryCache-2] Add BinaryDataFiberId prepare for bianry cache

### PR DESCRIPTION
**Pull request by LuciferYang**
## What changes were proposed in this pull request?

Add `BinaryDataFiberId`  prepare for bianry cache and rename `DataFiberId` to `VectorDataFiberId`.

## How was this patch tested?

Add BinaryFiberIdSuite

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala [hunk 1]
```diff
@@ -69,7 +69,8 @@ trait OapCache {
 
   protected def cache(fiber: FiberId): FiberCache = {
     val cache = fiber match {
-      case DataFiberId(file, columnIndex, rowGroupId) => file.cache(rowGroupId, columnIndex)
+      case binary: BinaryDataFiberId => binary.doCache()
+      case VectorDataFiberId(file, columnIndex, rowGroupId) => file.cache(rowGroupId, columnIndex)
```
Comment (zhixingheyi-tian) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala]: @LuciferYang 
When rename DataFiberId to VectorDataFiberId. 
Is this naming appropriate for OAP Fileformat?

Comment (LuciferYang) [reply src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala]: I think it's appropriate, `ColumnValues` also a Vector, but it's oap implementation, `ColumnValues` corresponds to `ColumnVector` and  `BatchColumn` corresponds to `ColumnarBatch`.

**Comment by zhixingheyi-tian** [comment]
@LuciferYang 
Can parquet binary cache support VectorizedParquetRecordReader ?

**Comment by LuciferYang** [comment]
Of course, all ParquetXXXReader will use ParquetFileReader  which BinaryCache worked

**Comment by LuciferYang** [comment]
@zhixingheyi-tian not available now, `VectorizedParquetRecordReader ` is spark internal code, but we can do this if we want it.

**Comment by zhixingheyi-tian** [comment]
@LuciferYang 
Sorry , write wrongly.
I mean, can this binary cache support  MrOapRecordReader ?

Or support both VectorizedOapRecordReader and MrOapRecordReader?

**Comment by LuciferYang** [comment]
@zhixingheyi-tian BinaryCache can support both VectorizedOapRecordReader and MrOapRecordReader :)

**Comment by zhixingheyi-tian** [comment]
@LuciferYang 
Hi , how many remaining small PRs?

**Comment by LuciferYang** [comment]
@zhixingheyi-tian 2 or 3

**Pull request merged**